app/src/main/java/org/septa/android/app/locationpicker/LineUnawareLocationPickerTabActivityHandler.java

```diff
@@ -147,6 +149,12 @@ public void onClick(View view) {
                     intent.putExtra(Constants.DESTINATION_STATION, destinationStation);
                     intent.putExtra(Constants.TRANSIT_TYPE, transitType);
 
+                    if (NextToArriveResultsActivity.class.equals(targetClass)) {
```

Comment:
Comparing classes is not safe.  Add a field to the constructor that will know where this is being used.